src/snapping/util.js

```diff
@@ -48,5 +53,40 @@ exports.getBufferLayer = (bufferLayerId, rootLayer, snapDistance) => {
   return bufferLayer;
 };
 
-exports.findVertexInCircle = (feature, circle) =>
+const findVertexInCircle = (feature, circle) =>
   getCoords(feature).find((coord) => pointInPolygon(coord, circle));
+
+exports.findVertexInCircleMulti = (snapGeom, circle, hoverPoint) => {
+  let vertex;
+  if (Array.isArray(snapGeom.geometry.coordinates[0][0])) {
+    const vertexesInCircle = snapGeom.geometry.coordinates
+      .flatMap((coords) =>
+        coords.map((coord) => {
+          if (coord[0]) {
+            const coordArray = Array.isArray(coord[0]) ? coord : [coord];
+            return findVertexInCircle(coordArray, circle);
+          }
+        })
+      )
+      .filter((value) => Array.isArray(value));
```

Comment:
This could use a come commenting or intermediate variable names too.

* Why are we checking `coord[0]`?
* Why do we `filter` for array results?

I think specifically, it's good to document what you expect the alternative to be in each of these cases. And are there turf utilities we could use to check the types of these geometries instead of checking these deeply-nested arrays? It's hard to follow which level of the array of array of arrays we're at sometimes.